### PR TITLE
Update functions and triggers

### PR DESCRIPTION
**Pull request by ponceta**
## General
 - [x] Fix a bug
 - [ ] Add a feature
 - [x] Maintenance / sustainability
 - [ ] Add Documentation

## Describe your changes

This aims to fix https://github.com/teksi/wastewater/issues/485

- [x] Rename symbology_on_reach_delete into modification_on_reach_delete
- [ ] Reorganize files in application (draft)

## Documentation

**Management functions :** 
- alter_symbology_triggers(enable/disable)
- checksymbology_triggers_enabled()
- tww_app.alter_modification_triggers(enable/disable)
- tww_app.check_modification_enabled()

**Modification functions:**
- tww_app.modification_on_reach_delete() (only on reach deletion)
-- Remove linked network element and reach points
-- Remove channel if no reach is left

**Symbology functions:**
 - tww_app.update_wastewater_node_symbology()
   - Updates selected nodes or all nodes, can be launch manually over the whole database or is triggered by 
     - a `channel` update, 
     - a `reach point` update, 
     - a `reach` update

## Issue ticket number and link
https://github.com/teksi/wastewater/issues/485

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] CI Tests are green
- [x] The documentation is up to date with the proposed change.
- [x] My work is ready for review

## Checklist before merge
- [x] A review has been performed
- [x] Comments are resolved
- [x] Documentation is ready

### CODE REVIEW

**Comment by ponceta** [comment]
@domi4484 @cymed @3nids could you review it :

- [ ] From the code organisation perspective (good / bad / what to adapt?)
- [ ] From the functions content (are the actual functions clear enough, should they be adapted into smaller pieces?)
- [ ] From the upgrade mechanics perspective (9A/9B)

This is already great because a single python instruction can rebuild the whole app and the whole app is cleaned by dropping cascade the app schema

image

**Review by domi4484 — APPROVED** [review]
@ponceta from the code perspective it looks good to me.
For the other questions I am missing some postgres expertise to anwer yet.
In any case whatever comes out from the other questions, this changes doesn't put any limit in future decisions and should be safe to merge

File: datamodel/app/triggers/create_triggers.sql [hunk 1]
```diff
@@ -0,0 +1,73 @@
+-- only update -> insert and delete are handled by reach trigger
```
Comment (3nids) [inline datamodel/app/triggers/create_triggers.sql:1]: I'm not a fan of separating the code which create the trigger from the ones which create the trigger functions.

**Review by 3nids — COMMENTED** [review]
Apart from the trigger file creation, this looks good to me.

I am not sure if I missed something important regarding this, but the main point is function renaming to clearly separate symbology from core netowrk business logic, is this correct?

File: datamodel/app/triggers/create_triggers.sql (hunk 1 above)
Comment (ponceta) [reply datamodel/app/triggers/create_triggers.sql]: I've brought back the triggers but organized respectively the functions with their triggers.

**Comment by ponceta** [comment]
@cymed if you can leave a comment on this I would very much appreciate! :)

**Comment by cymed** [comment]
I agree with 3nids on keeping triggers and trigger functions together. I like separating the base functions from the trigger functions and the triggers. 
As discussed in #438, we should also allow intermediate tables (although I don't know why we need these tables, a materialized view should do the job)

**Pull request merged**